### PR TITLE
Hrcpp 455/admin op

### PR DESCRIPTION
**Pull request by rigazilla**
Admin operation and tests

https://issues.jboss.org/browse/HRCPP-455

### CODE REVIEW

File: test/Infinispan.HotRod.Tests/RemoteCacheAdminTest.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        {
+            var ex = Assert.Throws<Infinispan.HotRod.Exceptions.HotRodClientException>(() =>
+            { remoteManager.Administration().CreateCache<string, string>("cache4AdminTest", "nonExistentTemplate"); });
+            StringAssert.Contains("ISPN000374:", ex.Message);
+        }
+
+        [Test]
+        public void getOrCreateWithoutTemplateTest()
+        {
+            remoteManager.Administration().GetOrCreateCache("default", (String)null);
```
Comment (mgencur) [inline test/Infinispan.HotRod.Tests/RemoteCacheAdminTest.cs]: I'm missing some assertions here. Perhaps something like in cacheCreateWithXMLConfigurationAndGetCacheTest

File: test/Infinispan.HotRod.Tests/RemoteCacheAdminTest.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        {
+            String cacheName = "cache4XmlCreateAndGetTest";
+            remoteManager.Administration().CreateCacheWithXml<object, object>(cacheName,
+               "<infinispan><cache-container><distributed-cache name=\""+cacheName+"\"/></cache-container></infinispan>");
+            var cache = remoteManager.GetCache<object, object>(cacheName);
+            Assert.NotNull(cache);
+        }
+
+        [Test]
+        [Ignore("Permanent doesn't work in windows")]
```
Comment (mgencur) [inline test/Infinispan.HotRod.Tests/RemoteCacheAdminTest.cs]: Why? could we have a JIRA for that?

File: src/Infinispan.HotRod/RemoteCacheManagerAdmin.cs [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+ 
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <typeparam name="K"></typeparam>
+        /// <typeparam name="V"></typeparam>
+        /// <param name="name"></param>
+        /// <param name="model"></param>
+        /// <returns></returns>
+        public IRemoteCache<K, V> CreateCache<K,V>(string name, string model)
```
[the remark this answers is not in the document]
Comment (rigazilla) [reply src/Infinispan.HotRod/RemoteCacheManagerAdmin.cs]: Sorry @mgencur I though I had polished this PR. Now it should be better.

File: test/Infinispan.HotRod.Tests/RemoteCacheAdminTest.cs (hunk 1 above)
Comment (rigazilla) [reply test/Infinispan.HotRod.Tests/RemoteCacheAdminTest.cs]: ok

File: test/Infinispan.HotRod.Tests/RemoteCacheAdminTest.cs (hunk 2 above)
Comment (rigazilla) [reply test/Infinispan.HotRod.Tests/RemoteCacheAdminTest.cs]: The issue is described here: https://issues.jboss.org/browse/HRCPP-468.

File: test/Infinispan.HotRod.Tests/RemoteCacheAdminTest.cs [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+            ClusterTestSuite.server1.ShutDownHotrodServer();
+            ClusterTestSuite.server2.ShutDownHotrodServer();
+            ClusterTestSuite.server1.StartHotRodServer();
+            ClusterTestSuite.server2.StartHotRodServer();
+            remoteManager = ClusterTestSuite.getRemoteCacheManager();
+            var ex = Assert.Throws<Infinispan.HotRod.Exceptions.HotRodClientException>(() =>
+            { remoteManager.Administration().CreateCache<string, string>(cacheName, "template"); });
+            StringAssert.Contains("ISPN000507:", ex.Message);
+        }
+
```
[the remark this answers is not in the document]
Comment (rigazilla) [reply test/Infinispan.HotRod.Tests/RemoteCacheAdminTest.cs]: I'll add this test with HRCPP-477, I would like to use getCacheNames to check the existence of a cache

**Comment by mgencur** [comment]
Looks good now. Thanks!

**Pull request merged**